### ISSUE TITLE
./configure: line 13402: syntax error near unexpected token `else'

### ISSUE THREAD

**Issue by paveljanik** [bitcoin-core/secp256k1#424]
Probably after merging #399, `autogen.sh` generates this code here:

```shell
  if test "$cross_compiling" = yes; then
  else
  cat >conftest.$ac_ext <<_ACEOF
```

ie. `else` immediately following `then`. On other systems, there is `:` (`true`) there.

@theuni Do you have an idea?

**Comment by gmaxwell** [comment]
This has been open forever without any other reports, I assume it got fixed. Feel free to reopen if otherwise.

**Comment by duhanov** [comment]
Same error!

**Comment by duhanov** [comment]
CentOS release 6.7 (Final)

**Comment by elichai** [comment]
I confirmed it in a docker, the problem is that CentOS 6 ships with an old autoconf, and after installing a third party update I was able to compile and pass the tests.
CentOS 6 got released a decade ago, stopped getting full updates 3 years ago and will be officially unmaintained in a few months.
The result of that is that I couldn't even clone this repo, because `yum install git` gave me 1.7.1, which github doesn't support, so I had to get git from a third party source.

So I think that not supporting CentOS 6 out of the box is quite reasonable seeing that even github doesn't support it out of the box (oh, and don't get me started on the SSL problems I had because all the CAs are out of date and it uses old openssl which I had to upgrade both)
